consensus/reactor.go

```diff
@@ -575,6 +575,16 @@ OUTER_LOOP:
 		if !peer.IsRunning() || !conR.IsRunning() {
 			return
 		}
+
+		isProposer := conR.conS.isProposer(conR.conS.privValidatorPubKey.Address())
+
+		// if we are the proposer wait for proposal to be complete before
+		// sending anything. This ensures block parts are distributed randomly.
+		if isProposer && !conR.conS.isProposalComplete() {
+			time.Sleep(10 * time.Millisecond)
```

Comment:
the delay here is completely arbitrary, and is soley to avoid a hot loop for each peer. if we're hyper optimizing, we would want this to be the lower bound of whatever the average time was for msgs to cross the internal queue.

Reply:
[question] do we expect to change this sleep time in the future? 

It seems unlikely but asking because if we do expect to change it, it may make sense to define a param for it so that a validator can override via config or CLI flag.

Reply:
There is already the `PeerGossipSleepDuration` config that can be used here as well.

Reply:
due to the nature of this timeout, the only thing it could change is the number of loops where we check if the block has been moved over the internal queue yet. While i haven't measured, given the loop is so simple I doubt there would be any measurable difference from .01ms to 100ms. Using an overly large value, such as one greater than the time it takes to ping a node, is probably the worst outcome, since we're waiting to gossip the block for no reason.